includes/class-attachments.php

```diff
@@ -0,0 +1,729 @@
+<?php
+/**
+ * Attachments processing file.
+ *
+ * @package Activitypub
+ */
+
+namespace Activitypub;
+
+use Activitypub\Collection\Posts;
+
+/**
+ * Attachments processor class.
+ */
+class Attachments {
+	/**
+	 * Directory for storing ap_post media files.
+	 *
+	 * @var string
+	 */
+	public static $ap_posts_dir = '/activitypub/ap_posts/';
+
+	/**
+	 * Directory for storing comment media files.
+	 *
+	 * @var string
+	 */
+	public static $comments_dir = '/activitypub/comments/';
+
+	/**
+	 * Initialize the class and set up filters.
+	 */
+	public static function init() {
+		\add_action( 'before_delete_post', array( self::class, 'delete_ap_posts_directory' ) );
+	}
+
+	/**
+	 * Delete the activitypub files directory for a post.
+	 *
+	 * @param int $post_id The post ID.
+	 */
+	public static function delete_ap_posts_directory( $post_id ) {
+		if ( Posts::POST_TYPE !== \get_post_type( $post_id ) ) {
+			return;
+		}
+
+		\WP_Filesystem();
+		global $wp_filesystem;
+
+		$activitypub_dir = self::get_storage_paths( $post_id, 'post' )['basedir'];
+
+		if ( $wp_filesystem->is_dir( $activitypub_dir ) ) {
+			$wp_filesystem->delete( $activitypub_dir, true );
+		}
+	}
+
+	/**
+	 * Import attachments from an ActivityPub object and attach them to a post.
+	 *
```

Comment:
I would at least mention that this is used by the importer!?

Reply:
Could you tell me more about that? We don't usually document callers, what makes this different?